### PR TITLE
introduction of customisation

### PR DESCRIPTION
**Pull request by valeriadamante**
This PR is made in order to integrate the possibility to customise the configurations for running the anatuple production

### CODE REVIEW

File: Common/anaTupleProducer.py [hunk 1]
```diff
@@ -186,6 +187,16 @@ def createAnatuple(inFile, outFile, config, sample_name, snapshotOptions,range,
     isData = False
     with open(args.configFile, 'r') as f:
         config = yaml.safe_load(f)
+    config_global = config['GLOBAL']
+    customisations= args.customisations.split(',')
```
Comment (kandrosov) [inline Common/anaTupleProducer.py:191]: `ApplyConfigCustomisations` is a general function, which we can use in different places. Put it in utilities
```
def ApplyConfigCustomisation(config_dict, customisations):
    ...
```

File: Common/anaTupleProducer.py [hunk 2]
```diff
@@ -186,6 +187,16 @@ def createAnatuple(inFile, outFile, config, sample_name, snapshotOptions,range,
...
+    for customisation in customisations:
+        substrings = customisation.split('=')
+        value = substrings[-1]
+        key_entries = substrings[0].split('.')
+        cfg_entry = config_global
+        for key in key_entries[:-1]:
+            cfg_entry = cfg_entry[key]
+        cfg_entry[key_entries[-1]] = value
```
Comment (kandrosov) [inline Common/anaTupleProducer.py:199]: convert `value` to have the same type as the original one.

File: Common/anaTupleProducer.py [hunk 3]
```diff
@@ -186,6 +187,16 @@ def createAnatuple(inFile, outFile, config, sample_name, snapshotOptions,range,
     isData = False
     with open(args.configFile, 'r') as f:
         config = yaml.safe_load(f)
+    config_global = config['GLOBAL']
+    customisations= args.customisations.split(',')
+    for customisation in customisations:
+        substrings = customisation.split('=')
```
Comment (kandrosov) [inline Common/anaTupleProducer.py:193]: perform validity checks here and below and raise error if something is not ok.

**Review by kandrosov — CHANGES_REQUESTED** [review]
resolve conflicts

File: Common/anaTupleProducer.py (hunk 1 above)
Comment (valeriadamante) [reply Common/anaTupleProducer.py]: ok

**Pull request closed without merging**